hub.jupytearth.org-image/Dockerfile

```diff
@@ -181,268 +181,12 @@ RUN export JUPYTER_DATA_DIR="$NB_PYTHON_PREFIX/share/jupyter" \
-# if you have a clear reason to not use conda.
-# https://github.com/pangeo-data/pangeo-docker-images/blob/master/pytorch-notebook/packages.txt
-#
-RUN echo "Installing pip packages..." \
- && export PATH=${NB_PYTHON_PREFIX}/bin:${PATH} \
- && pip install --no-cache-dir \
-        https://github.com/jupyterhub/jupyter-remote-desktop-proxy/archive/main.zip \
-            # /desktop part 5: jupyter-remote-desktop-proxy
-            #
-            # jupyter-remote-desktop-proxy enables us to visit the /desktop path
-            # just like we visit the /lab path. Visiting /desktop provides us
-            # with an actual remote desktop experience.
-            #
-            # NOTE: This package is not available on conda-forge, but available
-            #       on PyPI as jupyter-desktop-server I think but maybe not.
-            #
-            # NOTE: This install requires websockify to be installed via
-            #       conda-forge. We have also installed TurboVNC for performance
-            #       I think, and also various apt packages to get a desktop UI.
-            #
-        # https://github.com/minrk/jupyter-keepalive/archive/main.zip \
-            # This is a jupyter_server extension that is controllable via a
-            # JupyterLab plugin to keep a server running.
-            #
-            # ref: https://github.com/minrk/jupyter-keepalive
-            #
-            # NOTE: Disabled as we don't have nodejs installed, making us
-            #       require a pre-built wheel or installation of nodejs.
-        julia \
-            # To enable doing Julia stuff from Python
-            # ref: https://pyjulia.readthedocs.io/en/latest/index.html
-        jupyter-syncthing-proxy \
-            # We install the conda-forge package syncthing along with this, as
-            # this is just the glue to expose syncthing.
-            #
-            # ref, request:    https://github.com/pangeo-data/jupyter-earth/issues/103
-            # ref, source:     https://github.com/yuvipanda/jupyter-syncthing-proxy
-            # ref, dependency: https://anaconda.org/conda-forge/syncthing
-        plotly-geo \
-            # NOTE: This package is not available in conda (conda-forge or
-            #       plotly), even though they describe it to be.
-            # ref: https://github.com/plotly/plotly.py#extended-geo-support
-        # jupyter-datasette-proxy \
-            # FIXME: this is disabled because it malfunctions, and when it does,
-            #        all other jupyter-server-proxy processes fail and we
-            #        observe for example the vscode launcher entry in the
-            #        jupyterlab UI disappears.
-            # This package is not available in conda
- && echo "Installing pip packages complete!"
-
-
+# Install extra packages not present in base env
+ADD environment.yml /tmp/environment.yml
+RUN mamba env update -n ${CONDA_ENV} -y -f /tmp/environment.yml \
```

Comment:
This was previously `mamba install` but now its `mamba env update`, that may have a significant change in behavior right?

I presume this is like `pip install --upgrade --upgrade-strategy=eager` compared to `pip install`, which given that we have a lot of things in the base image may be a very big operation compared to `mamba install`.

Reply:
I don't know the details here and I don't want either of us to dig into the details, go for either if you guess it won't matter!